Assets/Mirror/Editor/NetworkScenePostProcess.cs

```diff
@@ -1,176 +1,72 @@
-                    right.RemoveAt(0);
-                }
-            }
-
-            // equal if both were empty or both had the same entry without any
-            // more children (should never happen in practice)
-            return 0;
-        }
-
-        public static int CompareNetworkIdentitySiblingPaths(NetworkIdentity left, NetworkIdentity right)
+        // sceneId challenges/requirements:
+        // * it needs to be 0 for prefabs
+        //   => we set it to 0 in NetworkIdentity.SetupIDs() if prefab!
+        // * it needs to be only assigned to objects that were in the scene
+        //   since the beginning
+        //   => OnPostProcessScene is only called once on load
+        // * there can be no duplicate ids
+        //   => Unity's fileID is unique
+        //   => fileID + sceneHash is potentially not unique across scenes, but
+        //      that risk is very small. we could still store fileId, hash in a
+        //      64 bit long if needed later.
+        // * duplicating the whole scene file should result in different
+        //   sceneIds for the files in the duplicated scene, even if that file
+        //   was never opened yet (e.g. when upgrading to Mirror)
+        //   => adding hash(scene.name) makes the fileIDs unique across scenes.
+        //   => this even works if the scene was never opened before, because
+        //      Unity calls OnPostProcessScene for all scenes when they are
+        //      built or opened
+        // * it needs to work with scenes that may or may not be in build index
+        //   and may or may not be enabled/disabled there.
+        //   => hash(scene.name) doesn't care
+        // * Ids need to be deterministic or saved if randomly generated
+        //   => saving ids is always difficult and full of edge cases
+        //   => generating a deterministic id in OnPostProcessScene is the
+        //      perfect(!) fail safe solution. doing this in OnValidate would be
+        //      very difficult to get right, especially for scenes that were
+        //      never opened (where OnValidate wasn't called yet).
+        static uint CalculateDeterministicSceneId(NetworkIdentity identity)
         {
-            return CompareSiblingPaths(SiblingPathFor(left.transform), SiblingPathFor(right.transform));
-        }
-
-        // we might have inactive scenes in the Editor's build settings, which
-        // aren't actually included in builds.
-        // so we have to only count the active ones when in Editor, otherwise
-        // editor and build sceneIds might get out of sync.
-        public static int GetSceneCount()
-        {
-#if UNITY_EDITOR
-            return EditorBuildSettings.scenes.Count(scene => scene.enabled);
-#else
-            return SceneManager.sceneCountInBuildSettings;
-#endif
+            // assign sceneId to hash(scene) + fileID
+            uint sceneHash = (uint)identity.gameObject.scene.name.GetStableHashCode();
```

Comment:
this is better than using the scene index,  because this works the same if scenes are reordered or not added to the list

Reply:
Why not hash the scene path? That way you can have duplicate scene names, but it's impossible to have two of the same file in the same path?  That will protect against two scenes with the same name in different folders.

Reply:
And as opposed to hashing the scene itself would allow the scene contents to be different but the hash value to stay the same. This would be useful for people who use headless build strippers.

Reply:
this PR was closed. valid points about hashing, but the GetFileID is not stable.